capstone/src/test/java/com/google/sps/objects/QuadTreeTest.java

```diff
@@ -0,0 +1,202 @@
+package com.google.sps.objects;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.gson.Gson;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.ArrayList;
+
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.Objects.QuadTree;
+import com.google.sps.Objects.BoundingBox;
+import com.google.sps.Objects.Location;
+import com.google.sps.Objects.Coordinate;
+
+/**
+ * Unit tests for QuadTree.
+ *
+*/
+public class QuadTreeTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig()
+              .setDefaultHighRepJobPolicyUnappliedJobPercentage(0));
+  
+  private static final Location LOC_1 = new Location("Test1", "address1", new Coordinate(40.48862915, -80.14141311));
+  private static final Location LOC_2 = new Location("Test2", "address2", new Coordinate(40.50170965, -80.09777729));
+  private static final Location LOC_3 = new Location("Test3", "address3", new Coordinate(40.57455069, -80.00992621)); 
+  private static final Location LOC_4 = new Location("Test4", "address4", new Coordinate(40.29651063, -79.98241256)); 
+  private static final Location LOC_5 = new Location("Test5", "address5", new Coordinate(40.49303469, -80.10254273)); 
+  private static final Location LOC_6 = new Location("Test6", "address6", new Coordinate(40.50245938, -80.08270203)); 
+  private static final Location LOC_7 = new Location("Test7", "address7", new Coordinate(40.4513275, -79.86228567)); 
+  private static final Location LOC_8 = new Location("Test8", "address8", new Coordinate(40.41427404,	-79.99527299)); 
```

Comment:
Personally, I would lean toward unrealistic, easy to follow values (40, -80), (40, -80.0000001), (40.0001, -80)

this will allow you to easily define the edge cases for distances where you can more confidently claim something is X far away, and hopefully end up with cleaner results that are easier to follow

Reply:
I'd also maybe consider instead of defining LOC's that you can't easily see IN the test themselves, a helper function:

```java
addTestLocation(x, y);
```

so you can easily see the coordinates you're comparing in the test and a developer at a glance should be able to tell something like (30, 30) is closer to (30, 40) than to (50, 30)